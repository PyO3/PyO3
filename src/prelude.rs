//! PyO3's prelude.
//!
//! The purpose of this module is to alleviate imports of many commonly used items of the PyO3 crate
//! by adding a glob import to the top of pyo3 heavy modules:
//!
//! ```
//! # #![allow(unused_imports)]
//! use pyo3::prelude::*;
//! ```

pub use crate::conversion::{FromPyObject, IntoPyObject};
#[allow(deprecated)]
pub use crate::conversion::{IntoPy, ToPyObject};
pub use crate::err::{PyErr, PyResult};
pub use crate::instance::{Borrowed, Bound, Py, PyObject};
pub use crate::marker::Python;
pub use crate::pycell::{PyRef, PyRefMut};
pub use crate::pyclass_init::PyClassInitializer;
pub use crate::types::{PyAny, PyModule};

#[cfg(feature = "macros")]
pub use pyo3_macros::{
    pyclass, pyfunction, pymethods, pymodule, FromPyObject, IntoPyObject, IntoPyObjectRef,
};

#[cfg(feature = "macros")]
pub use crate::wrap_pyfunction;
#[cfg(feature = "macros")]
#[allow(deprecated)]
pub use crate::wrap_pyfunction_bound;

pub use crate::types::any::PyAnyMethods;
pub use crate::types::weakref::PyWeakrefMethods;
