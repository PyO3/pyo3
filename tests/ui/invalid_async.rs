use pyo3::prelude::*;

#[pyfunction]
async fn check(){}

#[pyclass]
pub(crate) struct AsyncRange {
    count: i32,
    target: i32,
}
#[pymethods]
impl AsyncRange {
    async fn __anext__(mut _pyself: PyRefMut<'_, Self>) -> PyResult<i32> {
        Ok(0)
    }

    async fn foo(&self) {}
}

fn main() {}