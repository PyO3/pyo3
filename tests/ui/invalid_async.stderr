error: async functions are only supported with the `experimental-async` feature
 --> tests/ui/invalid_async.rs:4:1
  |
4 | async fn check(){}
  | ^^^^^

error: async functions are only supported with the `experimental-async` feature
  --> tests/ui/invalid_async.rs:13:5
   |
13 |     async fn __anext__(mut _pyself: PyRefMut<'_, Self>) -> PyResult<i32> {
   |     ^^^^^
